### DISCUSSION TITLE
Add `pyccel.lambdify` to accelerate SymPy expressions

### DISCUSSION THREAD

**Discussion by yguclu** [pyccel/pyccel#1333] [Ideas]
---

**Comment by EmilyBourne** [comment]
I think this already exists https://github.com/pyccel/pyccel/blob/master/pyccel/decorators.py#L26-L33. It's just that it's broken and never had any proper tests or documentation. Plus there were three approximately equivalent decorators for the same thing (`@python` (since removed), `@sympy` (broken), `@lambdify` (I have never seen an example of this)). See https://github.com/pyccel/pyccel/issues/330 , https://github.com/pyccel/pyccel/issues/692 , https://github.com/pyccel/pyccel/issues/668 

The `@sympy` decorator may be what you are looking for (e.g. see broken test https://github.com/pyccel/pyccel/blob/master/tests/symbolic/scripts/decorator.py). If not then maybe the handling of the `lambdify` function as envisaged here: https://github.com/pyccel/pyccel/blob/master/tests/symbolic/scripts/neural_net.py
